### ISSUE TITLE
Error in creating extension with spree -v 3.0.8

### ISSUE THREAD

**Issue by visheshell** [spree/spree#7237]
I faced with the error following the tutorial for creating extension.

- Ruby 2.1.8
- Rails 4.2.6
- Spree 3.0.8
- Windows 7

The output is the following:
```
$spree extension sample_sales
      create  spree_sample_sales
      create  spree_sample_sales/app
      create  spree_sample_sales/app/assets/javascripts/spree/backend/spree_sample_sales.js
      create  spree_sample_sales/app/assets/javascripts/spree/frontend/spree_sample_sales.js
      create  spree_sample_sales/app/assets/stylesheets/spree/backend/spree_sample_sales.css
      create  spree_sample_sales/app/assets/stylesheets/spree/frontend/spree_sample_sales.css
      create  spree_sample_sales/lib
      create  spree_sample_sales/lib/spree_sample_sales.rb
      create  spree_sample_sales/lib/spree_sample_sales/engine.rb
      create  spree_sample_sales/lib/spree_sample_sales/factories.rb
      create  spree_sample_sales/lib/generators/spree_sample_sales/install/install_generator.rb
      create  spree_sample_sales/bin
      create  spree_sample_sales/bin/rails
      create  spree_sample_sales/spree_sample_sales.gemspec
(erb):5:in `template': uninitialized constant SpreeCmd::Extension::Spree (NameError)
        from C:/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/erb.rb:850:in `eval'
        from C:/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/erb.rb:850:in `result'
        from C:/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/actions/file_manipulation.rb:116:in `block in template'
        from C:/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:53:in `call'
        from C:/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/actions/create_file.rb:53:in `render'
```
I believe the problem with `spree -v 3.0.8` because when i installed `spree -v 3.0.7` and then ran `spree _3.0.7_ extension sample_sales` everything worked fine and extension was created without any errors

**Comment by damianlegawiec** [comment]
Thanks for reporting it @visheshell - we'll take a look

**Comment by mdavo6** [comment]
I ran into this same error as well a couple of days ago.

When I commented out the 'context' variable in lines 113 and 116 of 'file_manipulation.rb' in Thor I was able to get through it

```
    def template(source, *args, &block)
      config = args.last.is_a?(Hash) ? args.pop : {}
      destination = args.first || source.sub(/#{TEMPLATE_EXTNAME}$/, "")

      source  = File.expand_path(find_in_source_paths(source.to_s))
      #context = instance_eval("binding")

      create_file destination, nil, config do
        content = ERB.new(::File.binread(source), nil, "-", "@output_buffer") #.result(context)
        content = block.call(content) if block
        content
      end
    end
```

However then ran into another error with 'Spree.version' not being defined (or something similar) in line 35 of extension.rb. When I commented this out as well I was able to create an extension with no errors.

Hope this helps.

**Comment by damianlegawiec** [comment]
Thanks @mdavo6 for additional information, this will help solve it out way faster

**Comment by mdavo6** [comment]
No problem Damian. Figure the least I could do was stop someone else wasting their life away getting to the same conclusion. Can't for the life of me work out why it is happening though. 

I'm enjoying working with Spree. Keep up the good work.

**Comment by jaykay** [comment]
I ran into the same issue. Though it shouldn't be related to the platform (I'm on Ubuntu 15.04, Ruby 2.2.3) I just wanted to add, that spree `master` eg. `3.1.0.beta`, produces the same issue.

I had to uninstall v3.0.8 before 3.0.7 worked, otherwise spree_cmd v3.0.8 got called somewhere in the stack. So, if using `spree _3.0.7_ extension simple_sales` does produce the same error, try `gem uninstall spree -v 3.0.8` and/or `gem uninstall spree_cmd -v 3.0.8` and run the generator again.
